code-challenges.test.js

```diff
@@ -14,38 +14,143 @@
+//    ✕ takes in a mixed data array and returns an array of only the REMAINDERS of the numbers when divided by 3
+//  ● threemainders › takes in a mixed data array and returns an array of only the REMAINDERS of the numbers when divided by 3
+//    ReferenceError: threemainders is not defined
 
 // b) Create the function that makes the test pass.
 
+//PSEUDOCODE
+//1. Create a function that takes in an array as a parameter
+//2. Declare a variable to store a filtered array
+//3. Filter out all values that aren't the number primitive data type and store them on the recently declared variable
+//4. Return the newly filtered array being mapped over with %3 performed on each iteration
+
+const threemainders = (array) => {
+  var filtered = array.filter(value => typeof value === 'number')
+  return filtered.map(value => value % 3)
+}
+
+// PASS  ./code-challenges.test.js
+//  threemainders
+//    ✓ takes in a mixed data array and returns an array of only the REMAINDERS of the numbers when divided by 3
+
+
+
+
 
 // --------------------3) Create a function that takes in an array of numbers and returns the sum of all the numbers cubed.
 
 // a) Create a test with an expect statement using the variables provided.
 
+describe("sumOfTheCubes", () => {
 const cubeAndSum1 = [2, 3, 4]
 // Expected output: 99
 const cubeAndSum2 = [0, 5, 10]
 // Expected output: 1125
-
+  it("takes in an array of numbers and returns the sum of all the numbers cubed", () => {
+    expect(sumOfTheCubes(cubeAndSum1)).toEqual(99)
+    expect(sumOfTheCubes(cubeAndSum2)).toEqual(1125)
+  })
+})
+
+// FAIL  ./code-challenges.test.js
+//  sumOfTheCubes
+//    ✕ takes in an array of numbers and returns the sum of all the numbers cubed
+//  ● sumOfTheCubes › takes in an array of numbers and returns the sum of all the numbers cubed
+//    ReferenceError: sumOfTheCubes is not defined
 
 // b) Create the function that makes the test pass.
+
+//PSEUDOCODE
+//1. Create a function that takes an array as a parameter
+//2. Declare a variable for adding the cubed values and storing the updated result
+//3. Map over the array where each iterated value is being cubed and added to the recently declared variable
+//4. Return the variable outside of the scope of the map
+
+const sumOfTheCubes = (array) => {
+  var sum = 0
+  array.map(value => {
+    sum += (value ** 3)
+  })
+  return sum
+}
+
+// PASS  ./code-challenges.test.js
+//  sumOfTheCubes
+//    ✓ takes in an array of numbers and returns the sum of all the numbers cubed
+
+//ALL TESTS
+// PASS  ./code-challenges.test.js
+//  profileArray
+//    ✓ takes in an array of objects and returns an array with a sentence about each person with their name capitalized
+//  threemainders
+//    ✓ takes in a mixed data array and returns an array of only the REMAINDERS of the numbers when divided by 3
+//  sumOfTheCubes
+//    ✓ takes in an array of numbers and returns the sum of all the numbers cubed
+//    Test Suites: 1 passed, 1 total
```

Comment:
Alright disregard my other comment about forgetting it test suite result. It located here. Good work